pkgs/servers/peertube/default.nix

```diff
@@ -0,0 +1,175 @@
+{ lib
+, callPackage
+, stdenv
+, fetchzip
+, youtube-dl
+, fetchurl
+, fetchFromGitHub
+, python
+, nodejs
+, nodePackages
+
+, ldap ? false
+}:
+
+let
+  fetchedGithub = path:
+    let
+      json = lib.importJSON path;
+    in rec {
+      version = json.tag;
+      name = "${json.meta.name}-${version}";
+      src = fetchFromGitHub json.github;
+    };
+
+  yarn2nixPackage = let
+    src = builtins.fetchGit {
+      url = "git://github.com/moretea/yarn2nix.git";
+      ref = "master";
+      rev = "780e33a07fd821e09ab5b05223ddb4ca15ac663f";
+    };
+  in
+    (callPackage src {}) // { inherit src; };
+
+  nodeHeaders = fetchurl {
+    url = "https://nodejs.org/download/release/v${nodejs.version}/node-v${nodejs.version}-headers.tar.gz";
+    sha256 = "16f20ya3ys6w5w6y6l4536f7jrgk4gz46bf71w1r1xxb26a54m32";
+  };
+
+  patchedPackages = stdenv.mkDerivation (fetchedGithub ./peertube.json // rec {
+    patches = if ldap then [ ./ldap.patch ././yarn_fix_bluebird_ldap.patch ] else [ ./yarn_fix_bluebird.patch ];
```

Comment:
This hack is a personnal one (I added ldap support to peertube because I needed it), it may not have its place here in a distribution package? (the patch was refused upstream)

Reply:
I agree.